### ISSUE TITLE
Exposure Logging is in "restricted" state even if it is active

### ISSUE THREAD

**Issue by kbobrowski** [corona-warn-app/cwa-documentation#262]
Note: this is a different issue to corona-warn-app/cwa-app-android#491 and can be solved independently.

## Describe the bug

When there is no internet access exposure logging will go to "restricted" state, the same state which appears when Bluetooth is off. But even without internet access phones will still receive / send Bluetooth beacons, and they can be matched later on with Diagnosis Keys (once internet access is enabled again).

This looks like sub-optimal UI/UX design, and leads to following issues:

- a person without mobile data enabled might believe that it does not make sense to take the phone or turn on Bluetooth when going e.g. grocery shopping, since exposure logging is not active anyway without internet access
- a person taking a flight might be led to believe that it does not make sense to turn on Bluetooth (after both internet access and Bluetooth have been turned off by airplane mode), since without internet the app won't work anyway (as a side note - it's possible to safely switch on Bluetooth after airplane mode was activated - some people may not know about this, perhaps it'd be good to indicate it in the app as well)
- **user has no way to opt-out (stop transmitting BLE beacons) once phone has internet connection disabled**

## Expected behaviour

It should be clear for the user that even without internet access app is working (it just needs internet access from time to time), and it's still worth to always carry a phone / keep Bluetooth on even in a situations without internet access.

**Additionally, user should be able to opt-out at any time, regardless of whether internet access is enabled, as stated in privacy policy:**

> To withdraw your consent to the exposure logging feature, you can disable the feature
using the toggle switch in the App or delete the App.

## Possible Fix

Better UI/UX design with two separate states:
- state indicating if exposure logging is on (this requires Bluetooth)
- state indicating if exposure risk score is up-to-date (this requires internet)

**Comment by aurisnoctis** [comment]
@pfille independently checked out in another issue if internet connection has an impact on bluetooth scanning:
> "In addition I checked, if CWA still works without an internet connection and it does. However, then in the app I get the status "Keine Internetverbindung", which is confusing as the tracing still works. Okay, in this case there is no notification."
https://github.com/corona-warn-app/cwa-app-android/issues/491#issuecomment-645524218

This test supports this issue and that it should be fixed as proposed by @kbobrowski.

**Comment by corneliusroemer** [comment]
Here are supporting pictures of the app's wrong message so that everyone knows what we're talking about.

I think this issue should get the *bug* label, not *enhancement* since it's clearly not expected/desired behaviour.
![Photo on 17-06-2020 at 21 20](https://user-images.githubusercontent.com/25161793/85020434-a02d5380-b170-11ea-9acb-b3b51ce1ba19.jpg)
![Photo on 17-06-2020 at 21 21](https://user-images.githubusercontent.com/25161793/85020440-a1f71700-b170-11ea-8eb1-cb48cd86ea7f.jpg)

**Comment by JayFoxRox** [comment]
> message should be:
"Can't download potential positives because of lack of internet connection. Bluetooth logging not affected."

I agree. 

---

I feel it's even worse for new users, because it shows (my phone is set up for american english):

> [red]**NO INTERNET CONNECTION** [WIFI "Bad" logo] [/red]
>
> [gray-box]
> **Risk unknown &rarr;**
> Since you have not activated
> exposure logging for long enough,
> we could not calculate your risk of
> infection.
> [/gray-box]

So this step does imply that something is wrong, or that it was never active in the first place.

When pressing the red message you see an even more confusing message:

> **Exposure Logging:** [Slider in Off position]
> Restricted  
>
> [gray-box]
> **Open Internet connection**
> Exposure logging requires an Internet connection to calculate
> exposures. Please turn on WIFI or mobile data in your device
> settings.
> [/gray-box]

**If I wasn't a developer with the ability to look up these issues, then I'd probably have uninstalled the app already, thinking I'd need a permanent internet connection.**

I try to minimize my connection to the internet to a minimum. My plan is to carry my phone with me at all times, and only check for infections every night (I hope I can manually force a check).

Given the confusing UI, I'm not sure:
- If the app works (I can't see if any beacons were ever exchanged, because a lack of timestamps when the last beacon was exchanged or even if I had ever seen another device)
- If it exchanges beacon despite lack of internet connection (UI implies it doesn't, because of the red message)
- If and when the internet was used to fetch a list (to me, UI implies this was never done, as it doesn't show any timestamp)
- How to control wether my phone sends beacons (to me, UI implies it currently doesn't do that, due to lack of internet connection)

So the UI should be improved to remove this uncertainty.

*For context: I have set up a phone specifically for corona-warn-app, and I run it without SIM (so no mobile internet plan or GSM), and for the most part without WiFi. I try to keep the cost and attack surface low (this is an Samsung S3 mini from 2012, running a custom ROM - LineageOS 14.1 (Android 7.1) by NovaFusion - that I don't fully trust, and OpenGApps Pico).*

**Comment by bufemc** [comment]
I agree to change the message. It have to be short, unambiguous, but it also have to be in a "non-nerd-language", so everybody understands. This might be too long, but goes in a good direction:
https://github.com/corona-warn-app/cwa-app-android/issues/610#issuecomment-645543087

**Comment by kbobrowski** [comment]
This issue may be related to issues discussed in #236 (see [this comment](https://github.com/corona-warn-app/cwa-documentation/issues/236#issuecomment-646238207)) - essentially **if internet connectivity is down in the short one-minute time window it may be that exposure status is not updated until next day**, and then again it has to be active in a short time window during next day.

If the situation described above is indeed the case, then it makes sense to **encourage users to always keep wifi / mobile internet on**, since if it is not the case exposure score might not necessarily be updated when internet connectivity is recovered. But it seems that this is very sub-optimal way of doing things, perhaps if #236 is resolved then it would be fine to tell users that they don't have to keep the internet on all the time.

Until #236 is resolved (especially the issue mentioned in linked comment), I would encourage everyone to **keep the internet on at all times**, it's true that Bluetooth beacons will be exchanged without the internet, but the status of the exposure might not be updated in the background (or when internet connectivity is recovered).

**Comment by mwbaumstark** [comment]
As increasing the limits for the server connection seem complicated: Could one for the time being
- store a timestamp of the last successful server connection
- initiate the exchange routine every hour
- check within the routine whether a successful download exists within the last 24 h
- if TRUE stop
- ELSE check for Internet connection
- if TRUE do the exchange with the server
 ELSE stop
This would at least solve the problem of postponing the exchange for one day if it fails today.

**Comment by corneliusroemer** [comment]
Dear maintainers, I don't want too seem pestering, I know you have an awful lot to do right now. But given that this may be one of the top 5 community raised issues in terms of severity/impact, I would very much appreciate if you could comment on the discussion so far and give your viewpoint after 3 days. 
I also reiterate my suggestion that this issue be labelled a "bug" instead of "enhancement" as it clearly is one.
Also, may I ask why no one has been assigned to this issue yet? Not entirely sure what this means for your workflow, but it can't harm to have one person responsible - you are assigning frequently on less severe issues.
Tagging @tkowark @SebastianWolf-SAP @mynchau @jakobmoellersap as those who are most active commenting/closing issues.

**Comment by SebastianWolf-SAP** [comment]
Dear @kbobrowski, dear contributors,

sorry that it might have taken a little too long for you,. We've received a response from our UX colleagues:

- The risk status card is shown even if internet connectivity is not available, even if new risk calculations are not possible. The error information itself about the lack of internet connectivity is limited to the exposure logging status element (which it is impacting directly) 
- In a future version we will clarify the user guidance and the setting regarding exposure logging based on representative end-user testing for the situation if internet connection is not available (thank you for the input)

As a consequence, we will take this issue to our newly created cwa-backlog repository. That repository hosts features or bugfixes that are planned, but can't be implemented directly or simply take some time.

The issue will move back to a development repository once the end-user testing is done, an agreement with the contracting parties about the implementation has been reached that it's due to be implemented shortly.

Best regards,
SW
Corona-Warn-App Open Source Team
